From: Debian Cyrus Team <pkg-cyrus-imapd-debian-devel@lists.alioth.debian.org>
Date: Tue, 4 Aug 2015 10:38:04 +0200
Subject: 03-fix_docs.dpatch

#! /bin/sh /usr/share/dpatch/dpatch-run
_# 03-fix_docs.dpatch by Sven Mueller <debian@incase.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix FAQ and virtual domains installation docs

@DPATCH@
---
 doc/faq.html                 |  2 +-
 doc/install-virtdomains.html |  4 ++++
 lib/imapoptions              | 11 +++++++++--
 3 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/doc/faq.html b/doc/faq.html
index 15ec74e..e639d32 100644
--- a/doc/faq.html
+++ b/doc/faq.html
@@ -168,7 +168,7 @@ Dec  6 12:58:57 mail3.andrew.cmu.edu imapd[1297]: cannot connect to saslauthd se
 </pre>
 
 <p>Make sure that <tt>saslauthd</tt> is running and that the cyrus
-user can access the unix domain socket (defaults to <tt>/var/run/mux</tt>).</p>
+user can access the unix domain socket (defaults to <tt>/var/run/saslauthd/mux</tt>).</p>
 
 </dd>
 
diff --git a/doc/install-virtdomains.html b/doc/install-virtdomains.html
index 957e26c..9d6da47 100644
--- a/doc/install-virtdomains.html
+++ b/doc/install-virtdomains.html
@@ -101,6 +101,10 @@ then set <tt>defaultdomain</tt> to 'xxx.yyy.zzz'.
 to the <tt>defaultdomain</tt>.  See <a href="#admins">Administrators</a>
 below for further discussion.
 
+<p><b>Note:</b> Users in the <tt>defaultdomain</tt> must <i>not</i>
+log in using the <tt>user@domain</tt> cross-realm login syntax. The
+<tt>user@domain</tt> type specification only works for them in LMTP.
+
 <p>Here is a sample <tt>imapd.conf</tt> with a minimal set of configuration
 options.</p>
 
diff --git a/lib/imapoptions b/lib/imapoptions
index 59990f7..12ab9b5 100644
--- a/lib/imapoptions
+++ b/lib/imapoptions
@@ -525,7 +525,11 @@ Blank lines and lines beginning with ``#'' are ignored.
    mailbox that does not have a parent mailbox. */
 
 { "defaultdomain", NULL, STRING }
-/* The default domain for virtual domain support */
+/* The default domain for virtual domain support. Note that this domain
+   is stripped from the email-address transmitted using LMTP, but it
+   is not stripped from usernames at login-time. For imapd/pop3d, 
+   "user" and "user@defaultdomain" specify two different users.
+   Please check install-virtdomains.html for details. */
 
 { "defaultpartition", NULL, STRING }
 /* The partition name used by default for new mailboxes.  If not
@@ -1049,7 +1053,10 @@ Blank lines and lines beginning with ``#'' are ignored.
 /* The list of remote realms whose users may authenticate using cross-realm
    authentication identifiers.  Separate each realm name by a space.  (A
    cross-realm identity is considered any identity returned by SASL
-   with an "@" in it.). */
+   with an "@" in it.) Note that to support multiple virtual domains
+   on the same interface/IP, you need to list them all as loginreals.
+   If you don't list them here, your users probably won't be able to
+   log in. */
 
 { "loginuseacl", 0, SWITCH }
 /* If enabled, any authentication identity which has \fBa\fR rights on a
